source/gui/settingsDialogs.py

```diff
@@ -2879,6 +3051,332 @@ def onBlinkCursorChange(self, evt):
 	def onNoMessageTimeoutChange(self, evt):
 		self.messageTimeoutEdit.Enable(not evt.IsChecked())
 
+
+class VisionProviderStateControl(vision.providerBase.VisionProviderStateControl):
+	def __init__(
+			self,
+			parent: wx.Window,
+			providerInfo: vision.providerInfo.ProviderInfo
+	):
+		self._providerInfo = providerInfo
+		self._parent = parent
+
+	def startProvider(
+			self,
+	) -> None:
+		"""Initializes the provider in a way that is gui friendly,
+		showing an error if appropriate.
+		"""
+		success = True
+		initErrors = []
+		try:
+			vision.handler.initializeProvider(self._providerInfo)
+		except Exception:
+			initErrors.append(self._providerInfo.providerId)
+			log.error(
+				f"Could not initialize the {self._providerInfo.providerId} vision enhancement provider",
+				exc_info=True
+			)
+			success = False
+		if not success and initErrors:
+			if len(initErrors) == 1:
+				# Translators: This message is presented when
+				# NVDA is unable to load a single vision enhancement provider.
+				message = _("Could not load the {provider} vision enhancement provider").format(
+					provider=initErrors[0]
+				)
+			else:
+				initErrorsList = ", ".join(initErrors)
+				# Translators: This message is presented when
+				# NVDA is unable to load multiple vision enhancement providers.
+				message = _(f"Could not load the following vision enhancement providers:\n{initErrorsList}")
+			gui.messageBox(
+				message,
+				# Translators: The title of the vision enhancement provider error message box.
+				_("Vision Enhancement Provider Error"),
+				wx.OK | wx.ICON_WARNING,
+				self._parent
+			)
+
+	def terminateProvider(
+			self,
+			verbose: bool = False
+	) -> None:
+		"""Terminates one or more providers in a way that is gui friendly,
+		@verbose: Whether to show a termination error.
+		@returns: Whether initialization succeeded for all providers.
+		"""
+		terminateErrors = []
+		try:
+			# Terminating a provider from the gui should never save the settings.
+			# This is because termination happens on the fly when unchecking check boxes.
+			# Saving settings would be harmful if a user opens the vision panel,
+			# then changes some settings and disables the provider.

```

Comment:
Can the saveSettings parameter be removed entirely? Settings should only be saved when the rest
of the configuration is saved. The extension points for config can be used for that.
Perhaps there is a case I am not thinking of?

Reply:
If we're terminating a provider by hand, we should make sure that the settings are saved. I don't think that happens if we remove the parameter, right?

Reply:
Yes, you are right. I think the confusion here for me was thinking that this would save the config to the settings file. I wonder if we could rename this to make it clearer; `updateConfigSettings` perhaps?